proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -234,16 +221,11 @@ fieldWireType MessageField = FieldWireType Lengthy encodeMessage
                                 decodeMessage
 fieldWireType GroupField = GroupFieldType
 
--- TODO: check it's the same tag, and if not then fail the *whole* computation.
-endOfGroup :: String -> Int -> Parser (Maybe String)
+endOfGroup :: String -> Int -> Parser ()
 endOfGroup name tag = do
     TaggedValue tag' (WireValue wt _) <- getTaggedValue
     Equal <- equalWireTypes name EndGroup wt
-    return $ if tag == tag'
-                then Nothing
-                else Just $ "endOfGroup: unexpected tag " ++ show tag'
-                                            ++ " for field " ++ show name
-                                                ++ ", expected " ++ show tag
+    guard (tag == tag')
```

Comment:
I tried it and it doesn't affect the final error message since the outer loop `parseMessage` will always just keep going when `end` fails.  Instead, it fails in `parseAndAddField` the next time it tries to parse a tag.  This is a change from the previous logic of this PR that was able to immediately fail when this guard failed.

I think on balance the simpler code is better, especially since groups are deprecated, and we may eventually move away from attoparsec to a parser library with more limited (or no) backtracking.